service/matching/taskReader.go

```diff
@@ -158,58 +185,61 @@ getTasksPumpLoop:
 						// This indicates the task list may have moved to another host.
 						tr.tlMgr.Stop()
 					} else {
-						tr.tlMgr.logger.Error("Persistent store operation failure",
+						tr.logger.Error("Persistent store operation failure",
 							tag.StoreOperationUpdateTaskList,
 							tag.Error(err))
 					}
 					// keep going as saving ack is not critical
 				}
 				tr.Signal() // periodically signal pump to check persistence for tasks
-				updateAckTimer = time.NewTimer(tr.tlMgr.config.UpdateAckInterval())
+				updateAckTimer = time.NewTimer(tr.config.UpdateAckInterval())
 			}
 		case <-checkIdleTaskListTimer.C:
 			{
 				if tr.isIdle(lastTimeWriteTask) {
 					tr.handleIdleTimeout()
 					break getTasksPumpLoop
 				}
-				checkIdleTaskListTimer = time.NewTimer(tr.tlMgr.config.IdleTasklistCheckInterval())
+				checkIdleTaskListTimer = time.NewTimer(tr.config.IdleTasklistCheckInterval())
 			}
 		}
-		scope := tr.tlMgr.scope.Tagged(getTaskListTypeTag(tr.tlMgr.taskListID.taskType))
-		scope.UpdateGauge(metrics.TaskBacklogPerTaskListGauge, float64(tr.tlMgr.taskAckManager.GetBacklogCount()))
+		scope := tr.scope.Tagged(getTaskListTypeTag(tr.taskListID.taskType))
+		scope.UpdateGauge(metrics.TaskBacklogPerTaskListGauge, float64(tr.taskAckManager.GetBacklogCount()))
 	}
 
 	updateAckTimer.Stop()
 	checkIdleTaskListTimer.Stop()
 }
 
 func (tr *taskReader) getTaskBatchWithRange(readLevel int64, maxReadLevel int64) ([]*persistence.TaskInfo, error) {
-	response, err := tr.tlMgr.executeWithRetry(func() (interface{}, error) {
-		return tr.tlMgr.db.GetTasks(readLevel, maxReadLevel, tr.tlMgr.config.GetTasksBatchSize())
-	})
+	var response *persistence.GetTasksResponse
+	op := func() (err error) {
+		response, err = tr.db.GetTasks(readLevel, maxReadLevel, tr.config.GetTasksBatchSize())
+		return
+	}
+	err := tr.throttleRetry.Do(context.Background(), op)
```

Comment:
this `throttleRetry.Do` appears to be semantically different than the `tr.tlMgr.executeWithRetry` function in that it isn't - unless I'm confused - handling conditional failures. 

Do you want to make this change in the refactor?

Reply:
GetTasks doesn't return conditional failures

Reply:
🤦  yes